### PR TITLE
azure: mention that we search azure config

### PR DESCRIPTION
**Pull request by isidentical**
> You may disregard these recommendations if you used the **Edit on GitHub** button from dvc.org to improve a doc in place.

❗ Please read the guidelines in the [Contributing to the Documentation](https://dvc.org/doc/user-guide/contributing/docs) list if you make any substantial changes to the documentation or JS engine.

🐛 Please make sure to mention `Fix #issue` (if applicable) in the description of the PR. This causes GitHub to close it automatically when the PR is merged.

Please choose to [allow us](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork) to edit your branch when creating the PR.

Thank you for the contribution - we'll try to review it as soon as possible. 🙏

See: #2485

### CODE REVIEW

File: content/docs/command-reference/remote/modify.md [hunk 1]
```diff
@@ -444,6 +444,11 @@ order of precedence):
 
 Note that Azure remotes can also authenticate via environment variables (instead
 of `dvc remote modify`). These are tried if none of the params above are set.
+Also some options can be propagated from the azure config file (which is managed
+through
+[az config](https://docs.microsoft.com/en-us/cli/azure/config?view=azure-cli-latest)).
+The default directory where the config will be searched is `~/.azure` but it can
```
Comment (jorgeorpinel) [inline content/docs/command-reference/remote/modify.md:450]: Questions:
- Which options can come from an Azure config file?
- Will remote config params take precedence over Azure config file options?

Comment (isidentical) [reply content/docs/command-reference/remote/modify.md:450]: > Which options can come from an Azure config file?

`connection_string`, `account_name`, `account_key`, `sas_token` and `container_name

> Will remote config params take precedence over Azure config file settings?
> Will env vars take precedence over Azure config file settings?

Yes, and yes. Precedence order: `dvc config > environment variables > azure config`

File: content/docs/command-reference/remote/modify.md [hunk 2]
```diff
@@ -481,6 +481,12 @@ $ export AZURE_PASSWORD='mysecret'
 > [env vars](https://docs.microsoft.com/en-us/python/api/azure-identity/azure.identity.environmentcredential)
 > available.
 
+Also some options can be propagated from the azure config file (which is managed
```
Comment (shcheklein) [inline content/docs/command-reference/remote/modify.md:484]: Question: `some options` - what does it mean?

-------

I think we should describe all possible source and their priorities at the very beginning of the section. No chances people will get to this last paragraph. And if I understand this correctly (this files serves the same role as AWS credentials), this seems to be an important information that we want and should mention from the very beginning. 

Something like: there are a few ways to define config values, in the order of their priority:

1. config files
2. env
3. DVC config


etc 


(This is just an example, I don't know the details here)

Comment (isidentical) [reply content/docs/command-reference/remote/modify.md:484]: > Question: some options - what does it mean?

https://github.com/iterative/dvc.org/pull/2703#discussion_r689458522

Comment (shcheklein) [reply content/docs/command-reference/remote/modify.md:484]: Thanks! That comment if you put it somewhere at the top of the document would be better - it's explicit, it explains the order clearly.

File: content/docs/command-reference/remote/modify.md (hunk 1 above)
Comment (jorgeorpinel) [reply content/docs/command-reference/remote/modify.md]: Thanks for the info. I'll take it from here 👍

File: content/docs/command-reference/remote/modify.md (hunk 2 above)
Comment (jorgeorpinel) [reply content/docs/command-reference/remote/modify.md]: > we should describe all possible source and their priorities at the very beginning

Agree. Done in 8b035c12. But it's gotten really complicated. This needs it's own guide (see https://github.com/iterative/dvc.org/issues/1859, it keeps popping up so hopefully we can prioritize it).

**Review by shcheklein — APPROVED** [review]

**Pull request merged**